src/main/io/gps.h

```diff
@@ -171,7 +365,7 @@ extern gpsData_t gpsData;
 extern gpsSolutionData_t gpsSol;
 
 #define GPS_SV_MAXSATS_LEGACY   16U
-#define GPS_SV_MAXSATS_M8N      32U
+#define GPS_SV_MAXSATS_M8N      48U
```

Comment:
This is probably largest message. Is it really necessary?

Reply:
Thanks, and to be truthful, I don't know for sure.
The GPS_SV_MAXSATS_M8N value doesn't affect the size of the NAV-SAT message that we receive, but does modify what we do with it. 
When set to 48, we will see Satellite information in Configurator, even if we have more than 32 satellites in view.  But if set to the previous value of 32, and we have more than 32 sats in view, the Sat information won't appear in Configurator.
I don't know why the value must be more than 32 to work in such a situation; it seems like a bug, but I don't know where the bug is.

Reply:
It is also included `_buffer` union and is used to defined UBLOX_PAYLOAD_SIZE (which is wrong, sizeof union would be cleaner). Larger frames are ignored
https://github.com/betaflight/betaflight/blob/f3a5171307d50c7a98306439146277704dde8fdd/src/main/io/gps.c#L2349-L2351

This can be quite easily refactored:
 - code already skips writing of overflow data, just change test to `sizeof`
 - check for reasonable packet size (UBLOX_PAYLOAD_SIZE_ACCEPTED ?)
 - (maybe) record that packet was truncated for UBLOX_parse_gps.
 - (better) size `_buffer` union members to reasonable size and use min(configured size, received size) when processing arrays

Reply:
I think this problem has been resolved.  @ZzyzxTek found some lines of code that rejected the sat info string if it was larger than the allowed payload size.  Removing these solved the problem, and allows us to re-set the payload size back to that for 32 sats.  If more data arrives than that, it is ignored.  I can confirm this works properly, and appears to have no adverse consequences.

Reply:
Maybe, with this fix in place, we can reduce GPS_SV_MAXSATS_M8N even more, up to next largest message supported. But not in this PR ...